### PR TITLE
more fixes on dup indexes, more tests around that

### PR DESCRIPTION
**Pull request by earonesty**
new index thing could create dups

more tests around that

warn if existing model has dups

### CODE REVIEW

File: notanorm/base.py [hunk 1]
```diff
@@ -680,13 +680,25 @@ def create_model(self, model: DbModel, ignore_existing=False):
...
-    ):  # pragma: no cover
-        raise RuntimeError("Generic create table not supported")
+    @abstractmethod
+    def create_table(self, name, schema: DbTable, ignore_existing=False):
+        ...
+
+    def create_indexes(self, name, schema: DbTable):
+        try:
+            existing = self.model()[name].indexes
+        except:
```
Comment (colatkinson) [inline notanorm/base.py:690]: [Remove bare except](https://stackoverflow.com/questions/54948548/what-is-wrong-with-using-a-bare-except). Don't think we want this to catch `KeyboardInterrupt`.

File: notanorm/sqlite.py [hunk 2]
```diff
@@ -218,7 +221,7 @@ def __info_to_model(cls, info):
 
     def model(self):
         """Get sqlite db model: dict of tables, each a dict of rows, each with type, unique, autoinc, primary"""
-        res = self.query("SELECT name, type from sqlite_master")
+        res = self.query("SELECT name, type from sqlite_master", no_capture=True)
```
Comment (colatkinson) [inline notanorm/sqlite.py:224]: I'm not sure how I feel about this. Getting the model is far from free, and we do it implicitly in a lot of places.

In many cases, I'd think you absolutely would want to capture that this happened, especially if it's somewhere perf sensitive.

File: notanorm/base.py [hunk 3]
```diff
@@ -739,13 +751,16 @@ def execute_ddl(self, sql: str, *dialect: str, ignore_existing=True):
...
 
-    def execute(self, sql: str, parameters=(), _script=False, write=True):
+    def execute(
+        self, sql: str, parameters=(), _script=False, write=True, no_capture=False
+    ):
         if "alter " in sql.lower() or "create " in sql.lower():
             self.__model_cache = None
 
-        self.__debug_sql(sql, parameters)
+        if not no_capture:
```
Comment (colatkinson) [inline notanorm/base.py:760]: IMO we should definitely log model lookups. Have used that to debug perf issues in the past.

**Review by colatkinson — APPROVED** [review]

File: notanorm/base.py (hunk 3 above)
Comment (earonesty) [reply notanorm/base.py:760]: removed

**Pull request merged**